File: php/src/Model/VoiceIceServers.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * **Experimental** — see {@see VoiceIceServer}.
 *
 * The answer to `voiceIce()` on either session plane.
 */
final class VoiceIceServers
{
    /** @param list<VoiceIceServer> $iceServers */
    public function __construct(public readonly array $iceServers = [])
    {
    }

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        /** @var list<VoiceIceServer> $servers */
        $servers = Json::parseEach($data, 'iceServers', [VoiceIceServer::class, 'fromArray']);
        return new self($servers);
    }
}

File: java/src/main/java/io/fivexer/sdk/model/VoiceIceServers.java
package io.fivexer.sdk.model;

import java.util.Collections;
import java.util.List;

/**
 * <strong>Experimental</strong> — see {@link VoiceIceServer}.
 *
 * <p>The answer to {@code voiceIce()} on either session plane.
 */
public class VoiceIceServers {

    private List<VoiceIceServer> iceServers;

    public List<VoiceIceServer> getIceServers() {
        return iceServers == null ? Collections.emptyList() : iceServers;
    }
}
